apps/frontend/src/modules/auth/auth-api.ts

```diff
@@ -20,6 +21,18 @@ class AuthApi extends BaseHTTPApi {
 		super({ baseUrl, http, path: APIPath.AUTH, storage });
 	}
 
+	public async getAuthenticatedUser(): Promise<null | UserAuthResponseDto> {
+		const response = await this.load(
+			this.getFullEndpoint(AuthApiPath.AUTHENTICATED_USER, {}),
+			{
+				hasAuth: true,
+				method: "GET",
+			},
+		);
+
+		return await response.json<null | UserAuthResponseDto>();
```

Comment:
it will return either user or error
```suggestion
		return await response.json<UserAuthResponseDto>();
```